dex/src/matching.rs

```diff
@@ -523,13 +536,15 @@ impl<'ob> OrderBookState<'ob> {
             }
         }
 
-        let net_fees_before_referrer_rebate = native_taker_fee - accum_maker_rebates;
-        let referrer_rebate = fees::referrer_rebate(native_taker_fee);
-        let net_fees = net_fees_before_referrer_rebate - referrer_rebate;
+        if owner != solana_program::system_program::ID.to_aligned_bytes() {
+            let net_fees_before_referrer_rebate = native_taker_fee - accum_maker_rebates;
+            let referrer_rebate = fees::referrer_rebate(native_taker_fee);
+            let net_fees = net_fees_before_referrer_rebate - referrer_rebate;
 
-        self.market_state.referrer_rebates_accrued += referrer_rebate;
-        self.market_state.pc_fees_accrued += net_fees;
-        self.market_state.pc_deposits_total -= net_fees_before_referrer_rebate;
+            self.market_state.referrer_rebates_accrued += referrer_rebate;
+            self.market_state.pc_fees_accrued += net_fees;
```

Comment:
I think the accounting here is likely not accurate. We can look through this more closely, but `SendTake` should certainly still have the taker apply fees

Reply:
I'm accounting for fees in process_send_take instead of here

Reply:
If you update `self.market_state.pc_deposits_total` prior to entering this function, does this prevent underflow from happening?

Reply:
Yes, we could. But we don't know how much to update pc_deposits_total by until after the new_bid function runs, right?

Reply:
I've modified the code to have the fee calculation stay in `matching.rs` but the deposits are adjusted in `send_take` function. I think this is ultimately safer